qiskit/transpiler/target.py

```diff
@@ -0,0 +1,600 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A target object represents the minimum set of information the transpiler needs
+from a backend
+"""
+
+from collections.abc import Mapping
+from collections import defaultdict
+import io
+import logging
+from typing import Union, Dict, Any
+
+import retworkx as rx
+
+from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
+from qiskit.transpiler.coupling import CouplingMap
+from qiskit.transpiler.instruction_durations import InstructionDurations
+from qiskit.transpiler.timing_constraints import TimingConstraints
+
+logger = logging.getLogger(__name__)
+
+
+class InstructionProperties:
```

Comment:
i think this class is very useful to attach properties on a per-instruction basis